build.gradle.kts

```diff
@@ -1,29 +1,135 @@
+            relativePath.isFile,
+            relativePath
+                .segments
+                .joinToString("_")
+                .replace("_action-types", "")
+        )
+    }
+    includeEmptyDirs = false
 }
 
-dependencies {
-    implementation("com.charleskorn.kaml:kaml:0.67.0")
-
-    testImplementation(platform("io.kotest:kotest-bom:5.9.1"))
-    testImplementation("io.kotest:kotest-runner-junit5")
-    testImplementation("io.kotest:kotest-assertions-core")
-}
+val schemaFile = file("src/main/schema/github-actions-typing.schema.json")
+val badDir = file("src/test/resources/bad-typings")
 
 kotlin {
     jvmToolchain(21)
+    jvm {
+        mainRun {
+            mainClass = "it.krzeminski.githubactionstyping.MainKt"
+        }
+
+        val test by testRuns.existing {
+            executionTask {
+                useJUnitPlatform()
+                configureInputsForSchemaTest()
+                systemProperty("schemaFile", schemaFile.absolutePath)
+                systemProperty("goodDir", collectGoodTypings.get().destinationDir.absolutePath)
+                systemProperty("badDir", badDir.absolutePath)
+            }
+        }
+    }
+
+    js {
+        useEsModules()
+        nodejs {
+            testTask {
+                configureInputsForSchemaTest()
+                environment("schemaFile", schemaFile.absolutePath)
+                environment("goodDir", collectGoodTypings.get().destinationDir.absolutePath)
+                environment("badDir", badDir.absolutePath)
+            }
+        }
+    }
+
+    sourceSets {
+        jvmMain {
+            dependencies {
+                implementation("com.charleskorn.kaml:kaml:0.67.0")
+            }
+        }
+
+        jvmTest {
+            dependencies {
+                runtimeOnly("org.junit.platform:junit-platform-launcher")
+                implementation(dependencies.platform("io.kotest:kotest-bom:5.9.1"))
+                runtimeOnly("io.kotest:kotest-runner-junit5")
+                implementation("io.kotest:kotest-framework-api")
+                implementation("io.kotest:kotest-framework-datatest")
+                implementation("io.kotest:kotest-assertions-core")
+
+                implementation("it.krzeminski:snakeyaml-engine-kmp:3.1.0")
+                implementation("io.github.optimumcode:json-schema-validator:0.3.1")
+            }
+        }
+
+        jsTest {
+            dependencies {
+                implementation(dependencies.platform("io.kotest:kotest-bom:5.9.1"))
+                implementation("io.kotest:kotest-framework-engine")
+                implementation("io.kotest:kotest-framework-api")
+                implementation("io.kotest:kotest-framework-datatest")
+                implementation("io.kotest:kotest-assertions-core")
+
+                implementation(kotlinWrappers.js)
+                implementation(kotlinWrappers.node)
+                implementation(kotlinWrappers.ajv)
+                implementation(kotlinWrappers.yaml)
```

Comment:
Why cannot we use https://github.com/OptimumCode/json-schema-validator and https://github.com/krzema12/snakeyaml-engine-kmp here as well? These are multiplatform libs.